tests/t319-basis.c

```diff
@@ -201,6 +201,85 @@ int main(int argc, char **argv) {
+          -0.022329099369260232, 0.022329099369260232},
+         {-0.083333333333333343, 0.083333333333333343, -0.31100423396407312, 0.31100423396407312, -0.022329099369260232, 0.022329099369260232,
+          -0.083333333333333343, 0.083333333333333343},
+         {-0.083333333333333343, 0.083333333333333343, -0.022329099369260232, 0.022329099369260232, -0.31100423396407312, 0.31100423396407312,
+          -0.083333333333333343, 0.083333333333333343},
+         {-0.022329099369260232, 0.022329099369260232, -0.083333333333333343, 0.083333333333333343, -0.083333333333333343, 0.083333333333333343,
+          -0.31100423396407312, 0.31100423396407312}                                                       },
+        {{-0.39433756729740643, 0, 0.39433756729740643, 0, -0.10566243270259357, 0, 0.10566243270259357, 0},
+         {-0.39433756729740643, 0, 0.39433756729740643, 0, -0.10566243270259357, 0, 0.10566243270259357, 0},
+         {-0.10566243270259357, 0, 0.10566243270259357, 0, -0.39433756729740643, 0, 0.39433756729740643, 0},
+         {-0.10566243270259357, 0, 0.10566243270259357, 0, -0.39433756729740643, 0, 0.39433756729740643, 0}},
+        {{-0.39433756729740643, 0, -0.10566243270259357, 0, 0.39433756729740643, 0, 0.10566243270259357, 0},
+         {-0.10566243270259357, 0, -0.39433756729740643, 0, 0.10566243270259357, 0, 0.39433756729740643, 0},
+         {-0.39433756729740643, 0, -0.10566243270259357, 0, 0.39433756729740643, 0, 0.10566243270259357, 0},
+         {-0.10566243270259357, 0, -0.39433756729740643, 0, 0.10566243270259357, 0, 0.39433756729740643, 0}}
+    };
+    CeedScalar *q_ref = NULL, *q_weights = NULL;
+
+    CeedBasisCreateTensorH1Lagrange(ceed, face_dim, 1, 2, P_1D, CEED_GAUSS, &basis_face);
+    CeedBasisCreateH1(ceed, CEED_TOPOLOGY_HEX, 1, 8, 4, (CeedScalar *)interp, (CeedScalar *)grad, q_ref, q_weights, &basis_cell_to_face);
+    CeedBasisCreateProjection(basis_cell_to_face, basis_face, &basis_proj);
+    const CeedScalar *interp_proj, *grad_proj;
+    const CeedScalar  interp_proj_ref[4][8] = {
+        {1, 0, 0, 0, 0, 0, 0, 0},
+        {0, 0, 1, 0, 0, 0, 0, 0},
+        {0, 0, 0, 0, 1, 0, 0, 0},
+        {0, 0, 0, 0, 0, 0, 1, 0}
+    };
+    /* clang-format off */
+    const CeedScalar grad_proj_ref[3][4][8] = {
+        {{-0.5, 0.5, 0,    0,   0,    0,   0,    0},
+         {0,    0,   -0.5, 0.5, 0,    0,   0,    0},
+         {0,    0,   0,    0,   -0.5, 0.5, 0,    0},
+         {0,    0,   0,    0,   0,    0,   -0.5, 0.5}},
+        {{-0.5, 0,   0.5,  0,   0,    0,   0,    0},
+         {-0.5, 0,   0.5,  0,   0,    0,   0,    0},
+         {0,    0,   0,    0,   -0.5, 0,   0.5,  0},
+         {0,    0,   0,    0,   -0.5, 0,   0.5,  0}},
+        {{-0.5, 0,   0,    0,   0.5,  0,   0,    0},
+         {0,    0,   -0.5, 0,   0,    0,   0.5,  0},
+         {-0.5, 0,   0,    0,   0.5,  0,   0,    0},
+         {0,    0,   -0.5, 0,   0,    0,   0.5,  0}}
+    };
```

Comment:
Style point that I could go either way on, but for the non-tensor element tests I put these matrices in a header

Reply:
done